e2e/e2e_test.go

```diff
@@ -160,6 +344,39 @@ func submitMintTransaction(setupEnv *setup.Setup, transactionResponse hedera.Tra
 	return res.Hash().String()
 }
 
+func sendEthTransaction(setupEnv *setup.Setup, asset string, t *testing.T) (common.Hash, *routerContract.RouterBurn) {
+	wrappedAsset, err := setup.WrappedAsset(setupEnv.Clients.RouterContract, asset)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println(fmt.Sprintf("Parsed [%s] to ETH Token [%s]", asset, wrappedAsset))
+
+	approvedValue := new(big.Int).SetUint64(receiveAmounts[asset])
+
+	approveTx, err := setupEnv.Clients.WHbarContract.Approve(setupEnv.Clients.KeyTransactor, setupEnv.ControllerContractAddress, approvedValue)
```

Comment:
`setupEnv.ControllerAddress` -> `setupEnv.Clients.Router.Controller(args)`. You will have to check for an error